File: 2132-convert-1d-array-into-2d-array/2132-convert-1d-array-into-2d-array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (m * n != original.size()) return {}; 

        vector<vector<int>> grid(m, vector<int>(n));
        for(int k = 0; k < m * n; k++){
            int i = k / n;  
            int j = k % n; 
            grid[i][j] = original[k];
        }
        return grid;
    }
};

File: 2132-convert-1d-array-into-2d-array/2132-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        if (original.length != m * n) {
            return new int[0][0]; 
        }
        
        int[][] target = new int[m][n];
        int row = 0;
        int col = 0;
        
        for (int i = 0; i < original.length; i++) {
            target[row][col++] = original[i];
            
            if (col == n) {
                row++;
                col = 0;
            }
        }
        
        return target;
    }
}
